### PR TITLE
Implement a mocked User entity and share it.

### PR DESCRIPTION
**Pull request by mshima**
- Simplifies future changes to User
- Simplifies relationships with User
- Starts calculating custom primary keys.

This is required for https://github.com/jhipster/generator-jhipster/issues/7112, https://github.com/jhipster/generator-jhipster/issues/12844
Simplifies implementation of https://github.com/jhipster/generator-jhipster/issues/11062, https://github.com/jhipster/generator-jhipster/issues/11760
Related to https://github.com/jhipster/generator-jhipster/issues/12374

---

Please make sure the below checklist is followed for Pull Requests.

-   [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.component.ts.ejs [hunk 1]
```diff
@@ -14,7 +14,7 @@ export default class <%=jhiPrefixCapitalized%>UserManagementView extends Vue {
       }
     });
   }
-  public init(userId: <% if (pkType === 'String'  || pkType === 'UUID') { %>string<% } else { %>number<% } %>): void {
+  public init(userId: <% if (user.primaryKeyType === 'String'  || user.primaryKeyType === 'UUID') { %>string<% } else { %>number<% } %>): void {
```
Comment (qmonmert) [inline generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.component.ts.ejs:17]: two spaces after `'String'` 😇

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs [hunk 2]
```diff
@@ -6,7 +6,7 @@ import {Authority} from '@/shared/security/authority'
 
 export default class UserManagementService {
 
-  public get(userId: <% if (pkType === 'String'  || pkType === 'UUID') { %>string<% } else { %>number<% } %>): Promise<any> {
+  public get(userId: <% if (user.primaryKeyType === 'String'  || user.primaryKeyType === 'UUID') { %>string<% } else { %>number<% } %>): Promise<any> {
```
Comment (qmonmert) [inline generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs:9]: same

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs [hunk 3]
```diff
@@ -18,7 +18,7 @@ export default class UserManagementService {
     return axios.put('api/users', user);
   }
 
-  public remove(userId: <% if (pkType === 'String'  || pkType === 'UUID') { %>string<% } else { %>number<% } %>): Promise<any> {
+  public remove(userId: <% if (user.primaryKeyType === 'String'  || user.primaryKeyType === 'UUID') { %>string<% } else { %>number<% } %>): Promise<any> {
```
Comment (qmonmert) [inline generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs:21]: same

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.component.ts.ejs (hunk 1 above)
Comment (mshima) [reply generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management-view.component.ts.ejs:17]:
```suggestion
  public init(userId: <% if (user.primaryKeyType === 'String' || user.primaryKeyType === 'UUID') { %>string<% } else { %>number<% } %>): void {
```

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs (hunk 2 above)
Comment (mshima) [reply generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs:9]:
```suggestion
  public get(userId: <% if (user.primaryKeyType === 'String' || user.primaryKeyType === 'UUID') { %>string<% } else { %>number<% } %>): Promise<any> {
```

File: generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs (hunk 3 above)
Comment (mshima) [reply generators/client/templates/vue/src/main/webapp/app/admin/user-management/user-management.service.ts.ejs:21]:
```suggestion
  public remove(userId: <% if (user.primaryKeyType === 'String' || user.primaryKeyType === 'UUID') { %>string<% } else { %>number<% } %>): Promise<any> {
```

**Review by murdos — APPROVED** [review]
Great idea!
I've not reviewed the PR in details, but I'm trusting you here @mshima

**Comment by mshima** [comment]
Thanks @murdos @DanielFran @qmonmert.

**Pull request merged**